src/main/java/com/searchclient/clientwrapper/service/SolrDocumentService.java

```diff
@@ -78,37 +88,88 @@ public SolrResponseDTO addDocuments(String collectionName, String payload, boole
 
 			if (!documentSatisfiesSchemaResponse.isObjectSatisfiesSchema()) {
 
-				// ERROR IN A DOCUMENT STRUCTURE
-				solrResponseDTO.setMessage(documentSatisfiesSchemaResponse.getMessage());
-				solrResponseDTO.setMessage("The JSON input document in the array doesn't satisfy the Schema. Error: "
-						+ documentSatisfiesSchemaResponse.getMessage());
-				solrResponseDTO.setStatusCode(400);
-				return solrResponseDTO;
-
+				return documentscemanotparse(solrResponseDTO, documentSatisfiesSchemaResponse);
 			}
-
 		}
 
 		// IF THE FLOW HAS REACHED HERE, IT MEANS THE INPUT JSON DOCUMENT SATISFIES THE
 		// SCHEMA.
 		// MicroserviceHttpGateway microserviceHttpGateway = new
 		// MicroserviceHttpGateway();
 
-		String url = baseMicroserviceUrl + inputDocumentMicroserviceAPI + "/" + collectionName;
+		String url = baseMicroserviceUrl + inputDocumentMicroserviceAPI + "/ingest-nrt/" + clientid + "/" + collectionName;
+
+		microserviceHttpGateway.setApiEndpoint(url);
+		microserviceHttpGateway.setRequestBodyDTO(payload);
+		String jsonString = microserviceHttpGateway.postRequestWithStringBody();
+
+		JSONObject jsonObject = new JSONObject(jsonString);
+		solrResponseDTO.setMessage(jsonObject.get("responseMessage").toString());
+		solrResponseDTO.setStatusCode((int) jsonObject.get("statusCode"));
```

Comment:
Suggestion 2: define constant for 'responseMessage' & 'statusCode' because these are members variables name of Response DTOs, so initialization has to be kept apart from logic